repositories/mentor_repository.js

```diff
@@ -0,0 +1,103 @@
+const { getKnex } = require("../db");
+const userRepo = require('./user_repository');
+const logger = require('../logger')(__filename);
+const { ROLE_NAMES } = require("../utils");
+
+async function getMentors(userId, status){
+  const knex = getKnex();
+  const allowedStatuses = await getEligibleMentorStatuses(userId);
+  const query = knex("mentors")
+    .join("users", "mentors.user_id", "users.id")
+    .select(
+      "mentors.id as mentor_id",
+      "mentors.user_id",
+      "mentors.about",
+      "mentors.status",
+      "mentors.contact",
+      "users.name",
+      "users.img as img_link",
+      "users.discord_nickname"
+    );
+  if (status) {
```

Comment:
same as in the `getEligibleMentorStatuses` method: this feels like too much of a hustle for a repository's method.
can we pass all necessary data as arguments and just build & process the query here?
also, i have a strong feeling that we might use `whereIn` for all cases.